File: tests/human/x/ex/cross_join_filter.ex
nums = [1, 2, 3]
letters = ["A", "B"]

pairs = for n <- nums, l <- letters, rem(n, 2) == 0 do
  %{n: n, l: l}
end

IO.puts("--- Even pairs ---")
Enum.each(pairs, fn p ->
  IO.puts("#{p.n} #{p.l}")
end)

File: tests/human/x/fs/cross_join_filter.fs
let nums = [1; 2; 3]
let letters = ["A"; "B"]

let pairs =
    [ for n in nums do
        for l in letters do
            if n % 2 = 0 then
                yield {| n = n; l = l |} ]

printfn "--- Even pairs ---"
for p in pairs do
    printfn "%d %s" p.n p.l

